From 6bbaf3b43011b1e42ab48b5f3877cd8bf1d0e6cd Mon Sep 17 00:00:00 2001
From: Pedro Palazon Candel <pedro@joyent.com>
Date: Wed, 20 Sep 2017 16:42:51 +0200
Subject: [PATCH] CAPI-535 runtests is incorrectly reporting success when test
 suite exits with non zero value Reviewed by: Trent Mick <trentm@gmail.com>
 Approved by: Trent Mick <trentm@gmail.com>

---
 test/runtests | 11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git a/test/runtests b/test/runtests
index 23e3359..6b9e1dd 100755
--- a/test/runtests
+++ b/test/runtests
@@ -18,10 +18,9 @@ if [ "$TRACE" != "" ]; then
     export PS4='${BASH_SOURCE}:${LINENO}: ${FUNCNAME[0]:+${FUNCNAME[0]}(): }'
     set -o xtrace
 fi
-# With this stuff enabled we'll get not summary at the end of the tests
-# execution if any test wrote to stderr.
-#set -o errexit
-#set -o pipefail
+
+set -o errexit
+set -o pipefail
 
 
 
@@ -112,10 +111,13 @@ if [[ -n "$opt_test_pattern" ]]; then
     echo "# Running filtered set of test files: $test_files"
 fi
 
+set +o errexit
 if [[ -n "$test_files" ]]; then
     PATH=$NODE_INSTALL/bin:$PATH TEST_CONFIG_FILE=$TOP/etc/config.json node test/test.js \
         | tee $OUTPUT_DIR/ufds.tap
 fi
+rv=$?
+set -o errexit
 
 echo ""
 echo "# test output:"
@@ -147,3 +149,4 @@ echo ""
 if [[ ${tests} != ${passed} ]]; then
     exit 1
 fi
+exit $rv
-- 
2.21.0

